tests/services/test_service.py

```diff
@@ -138,11 +139,24 @@ def test_missing_or_invalid_type(lang_data2, service, identity):
             ValidationError, service.create, identity, lang_data2)
 
 
-def test_read_all_no_cache(lang_type, lang_data_many, service, identity):
+def test_read_all_no_cache(
+    lang_type, lang_data_many, service, identity, cache
+):
     """ read_all method should return all languages created in this scope. """
-    items = service.read_all(identity, fields=[],
+    items = service.read_all(identity, fields=['id'],
                              type='languages', cache=False)
     assert set(lang_data_many).issubset(set([i.id for i in items._results]))
+    cached = current_cache.get("id")
+    assert not cached
+
+
+def test_read_all_cache(lang_type, lang_data_many, service, identity, cache):
+    """ read_all method should return all languages created in this scope. """
+    items = service.read_all(identity, fields=['id'],
+                             type='languages', cache=True)
+    assert set(lang_data_many).issubset(set([i.id for i in items._results]))
```

Comment:
minor: accessing `_result` is sort of breaking the intention of having it "private" (i.e. with `_`)... I am wondering we should just access `items` and use the dict approach:
```suggestion
    assert set(lang_data_many).issubset(set([i["id"] for i in items]))
```

---

I am also wandering since the `ResultItem` has an `.id` attribute, why does each element of `ResultItemList` not have it... WDYT? (With this I'm trying to trigger a discussion point, maybe there ia another issue to solve). For now, just the changes of what's written above 👼

Reply:
Each item in ResultItemList has an id, otherwise how this code will work?
> assert set(lang_data_many).issubset(set([i["id"] for i in items]))